File: Server/Models/Reminder.cs
using System.ComponentModel.DataAnnotations;

namespace Server.Models;

public class Reminder
{
    public int Id { get; set; }
    [StringLength(maximumLength: 50, MinimumLength = 1)]
    public string? Title { get; set; }
    public DateTime DateDue { get; set; }
    public Repeat? Repeat { get; set; }
    [StringLength(maximumLength: 1023, MinimumLength = 0)]
    public string? Notes { get; set; }
}

public enum Repeat
{
    // Null: 0
    None, // 1
    Daily, // 2
    Weekly, // 3
    Monthly, // 4
    Weekdays, // 5
    Weekends, // 6
    Yearly // 7
}

File: client/lib/data/models/reminder.dart
import 'dart:convert';


List<Reminder> reminderFromJson(String str) =>
  List<Reminder>.from(json.decode(str).map((x) => Reminder.fromJson(x)));

String reminderToJson(List<Reminder> data) =>
  json.encode(List<dynamic>.from(data.map((e) => e.toJson())));

class Reminder{
  final int? id;
  final String title;
  final DateTime dateDue;
  final Repeat repeat;
  final String notes;

  const Reminder({
    this.id,
    required this.title,
    required this.dateDue,
    required this.repeat,
    required this.notes,
  });

  factory Reminder.fromJson(Map<String, dynamic> json) => Reminder(
    id: json['id'],
    title: json['title'],
    dateDue: json['dateDue'],
    repeat: json['repeat'],
    notes: json['notes'],
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'title': title,
    'dateDue': dateDue,
    'repeat': repeat,
    'notes': notes,
  };
}

enum Repeat
{
  none,
  daily,
  weekly,
  monthly,
  weekdays,
  weekends,
  yearly,
}